pyccl/core.py

```diff
@@ -161,7 +178,9 @@ def __init__(
             baryons_power_spectrum='nobaryons',
             mass_function='tinker10',
             halo_concentration='duffy2008',
-            emulator_neutrinos='strict'):
+            emulator_neutrinos='strict',
+            background_on_input=False, chi_a=None, hoh0=None, a_array=None,
```

Comment:
Let's think of a homogeneous way of calling these variables. I'd suggest appending `_array` to all of them. Also, since the assumption is that all the arrays (`chi_a`, `hoh0`, `growth_array` and `fgrowth_array`) share the same `a_array`, I would put `a_array` before all the other ones in the function signature